ddsp/training/docker/user_interaction.py

```diff
@@ -1,84 +1,235 @@
-#!/usr/bin/env python3
-"""Script for gathering input, building the image and submitting job"""
+# Copyright 2020 The DDSP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Script for gathering input, building the image and submitting job.
+
+Usage:
+================================================================================
+You can specify all input using flags, then the script triggers building,
+pushing the image and submitting the job automatically.
+
+You can specify only a part of input using flags, then the script
+asks for missing parameters.
+
+The pip install installs a `ddsp_ai_platform`
+script that can be called directly.
+================================================================================
+
+ddsp_ai_platform \
+--data_path=gs://ddsp_training/data \
+--save_dir=gs://ddsp_training/model \
+--batch_size=128\
+--learning_rate=0.001
+
+
+"""
 
 import os
+import datetime
+import subprocess
+
+from absl import app
+from absl import flags
+from google.cloud import storage
+
+FLAGS = flags.FLAGS
+
+# Program flags.
+flags.DEFINE_string('data_path', '',
+                    'Path where the dataset for training is saved')
+flags.DEFINE_string('save_dir', '',
+                    'Path where checkpoints and summary events will be saved '
+                    'during training and evaluation.')
+flags.DEFINE_string('restore_dir', '',
+                    'Path from which checkpoints will be restored before '
+                    'training. Can be different than the save_dir.')
+flags.DEFINE_string('config_path', '',
+                    'Configuration file for AI Platform training')
+flags.DEFINE_string('project_id', '', 'GCP Project ID')
+flags.DEFINE_string('region', '', 'GCP region for running the training')
+flags.DEFINE_string('batch_size', '', 'Batch size')
+flags.DEFINE_string('learning_rate', '', 'Learning rate')
+flags.DEFINE_string('num_steps', '', 'Number of training steps')
+flags.DEFINE_string('steps_per_summary', '', 'Steps per summary')
+flags.DEFINE_string('steps_per_save', '', 'Steps per save')
+flags.DEFINE_string('early_stop_loss_value', '',
+                   'Early stopping. When the total_loss reaches below this '
+                   'value training stops.')
+
+def prompt(default, message, skip=False, arg='', local=False):
+  input_prompt = 'Insert '
```

Comment:
While technically correct, this is a lot harder to read / parse what's going on. I think you can make this clearer. Try using f-strings and making each component is complete and readable on it's own, so that there are less line-to-line dependencies and you can just read the code down to understand it.